bg_utils/__init__.py

```diff
@@ -60,32 +61,35 @@ def generate_config_file(spec, cli_args):
         YapconfLoadError: Missing 'config' configuration option (file location)
     """
     config = _generate_config(spec, cli_args)
+    config_file, config_type = _get_config_values(config)
 
-    if config.get('config', None):
-        spec._write_dict_to_file(config, config.config, 'json')
-    else:
-        print(config)
+    yapconf.dump_data(config.to_dict(), filename=config_file,
```

Comment:
Does this not just work with `yapconf.dump_data(config, ...)` ? You shouldn't have to convert it to a dictionary, since the object yapconf returns inhereits from dict.